### PR TITLE
strimzi-kafka-operator

### PR DESCRIPTION
**Pull request by developer-guy**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: strimzi-kafka-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    with:
+      repository: https://github.com/strimzi/strimzi-kafka-operator
+      tag: ${{package.version}}
+      expected-commit: 23c0fb0726e1f212c6182eb251ff23697e72a353
+
+  - runs: |
+      export JAVA_HOME="/usr/lib/jvm/java-17-openjdk"
+      # needs shellcheck
+      scversion="stable"
+      wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${scversion?}/shellcheck-${scversion?}.linux.${{build.arch}}.tar.xz" | tar -xJv
```
Comment (Dentrax) [inline strimzi-kafka-operator.yaml:38]: I was going to say "Should it be separate package?" but then noticed that we don't have `cabal` or `haskell` packaged in Wolfi. 🤔

File: strimzi-kafka-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+              if [[ $? -eq 0 ]]; then
+                  echo "Successfully unzipped $zipfile to $DEST_DIR"
+              else
+                  echo "Failed to unzip $zipfile"
+              fi
+          else
+              echo "No zip file found for pattern: ${file}-*-dist.zip"
+          fi
+      done
+
```
Comment (mamccorm) [inline strimzi-kafka-operator.yaml:86]: Did we try adding some tests?

Comment (mamccorm) [reply strimzi-kafka-operator.yaml:86]: I know some of these Java apps can be tricky to add melange test coverage for, but it might be possible

Comment (developer-guy) [reply strimzi-kafka-operator.yaml:86]: I added some test but I'm not sure I could go further since the app is failing while detecting the k8s version, I noted it to the tests as a comment.

Comment (developer-guy) [reply strimzi-kafka-operator.yaml:86]: I found similar issues in the past:

- https://github.com/strimzi/strimzi-kafka-operator/issues/6848
- https://github.com/strimzi/strimzi-kafka-operator/issues/2920

File: strimzi-kafka-operator/launch_java.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Default memory options used when the user didn't configured any of these options, we set the defaults
+if [[ "$JAVA_OPTS" != *"MinRAMPercentage"* && "$JAVA_OPTS" != *"MaxRAMPercentage"* ]]; then
+  JAVA_OPTS="${JAVA_OPTS} -XX:MinRAMPercentage=15 -XX:MaxRAMPercentage=20"
+fi
+
+# Disable FIPS if needed
+if [ "$FIPS_MODE" = "disabled" ]; then
+    JAVA_OPTS="${JAVA_OPTS} -Dcom.redhat.fips=false"
+fi
```
Comment (mamccorm) [inline strimzi-kafka-operator/launch_java.sh:34]: Do we need this? seems unrelated to this type of package?

Comment (developer-guy) [reply strimzi-kafka-operator/launch_java.sh:34]: I copied it directly from here: https://github.com/strimzi/strimzi-kafka-operator/blob/main/docker-images/operator/scripts/launch_java.sh

Comment (developer-guy) [reply strimzi-kafka-operator/launch_java.sh:34]: do you want me to patch this line to remove it?

File: strimzi-kafka-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      - busybox
+      - ca-certificates-bundle
+      - coreutils
+      - curl
+      - docker # not gonna be needed but their script requires docker to be installed
+      - helm
+      - make
+      - maven
+      - openjdk-17
+      - openjdk-17-default-jvm
```
Comment (EyeCantCU) [inline strimzi-kafka-operator.yaml:22]: `openjdk-17` will pull in the JVM

File: strimzi-kafka-operator.yaml [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  github:
+    identifier: strimzi/strimzi-kafka-operator
+
+test:
+  environment:
+    contents:
+      packages:
+        - posix-libc-utils
+        - bash
+        - openjdk-17
```
Comment (EyeCantCU) [inline strimzi-kafka-operator.yaml:98]: We just need the JVM/JRE for tests

File: strimzi-kafka-operator.yaml (hunk 1 above)
Comment (EyeCantCU) [reply strimzi-kafka-operator.yaml:39]: Not crazy about this. Have we explored packaging it?

**Comment by developer-guy** [comment]
I closed the tests since tests are successfully when I try locally but fails on CI

**Comment by developer-guy** [comment]
kindly ping for one more final review folks @EyeCantCU @mamccorm

File: strimzi-kafka-operator.yaml (hunk 2 above)
Comment (Dentrax) [reply strimzi-kafka-operator.yaml:86]: Maybe we can just add a failing test that do something like: `grep Detection of Kuberetes version failed`?

Comment (developer-guy) [reply strimzi-kafka-operator.yaml:86]: I removed the tests due to inconsistency between local and CI, I tested this package by creating an image locally, so, it should be bullet-proof 🔫

File: strimzi-kafka-operator.yaml (hunk 1 above)
Comment (jamonation) [reply strimzi-kafka-operator.yaml:39]: Sadly cabal and haskell are not bootstrappable from source (many have tried in the haskell community but the original sources and compiler are lost). It would take a larger foundations effort to get those into extra-packages in a reasonably verifiable manner, which is beyond what we can do here.

Comment (jamonation) [reply strimzi-kafka-operator.yaml:39]: All that said, can this checkout use the git fetch pipeline? Or at least checkout a commit hash? Failing those, shasum verification of the tarball?

**Review by EyeCantCU — CHANGES_REQUESTED** [review]
I know we weren't able to have a functional test here, but could we do some kind of sanity check? Make sure things are in the right place, etc? Also, see JDK feedback

**Comment by developer-guy** [comment]
the build breaks when I remove the openjdk-17-default-jvm dependency :/

File: strimzi-kafka-operator.yaml (hunk 1 above)
Comment (EyeCantCU) [reply strimzi-kafka-operator.yaml:39]: @developer-guy see above, can we get the `fetch` pipeline implemented here?

Comment (developer-guy) [reply strimzi-kafka-operator.yaml]: it looks a bit ugly but yeah I made it finally 🫠

File: strimzi-kafka-operator.yaml [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+update:
+  enabled: true
+  github:
+    identifier: strimzi/strimzi-kafka-operator
+
+test:
+  pipeline:
+    - name: check folders are not empty
+      runs: |
+        #!/bin/bash
```
Comment (EyeCantCU) [inline strimzi-kafka-operator.yaml:117]: This doesn't work in tests. We'd do something like:

````
bash -c '
test contents
'
````

Comment (developer-guy) [reply strimzi-kafka-operator.yaml:117]: actually we don't need this one, let's remove it

**Review by EyeCantCU — APPROVED** [review]
